components/raftstore/src/store/fsm/store.rs

```diff
@@ -826,28 +797,69 @@ impl<EK: KvEngine, ER: RaftEngine, T: Transport> PollHandler<PeerFsm<EK, ER>, St
     }
 
     fn end(&mut self, peers: &mut [Box<PeerFsm<EK, ER>>]) {
-        self.flush_ticks();
+        let now = TiInstant::now();
+        if self.poll_ctx.trans.need_flush()
+            && now.saturating_duration_since(self.last_flush_msg_time)
```

Comment:
What if it fails the check and then the system becomes quiet for a long time? When will the trans be flushed?

Reply:
If so, `pause` should be called and the `flush` will be called in here.